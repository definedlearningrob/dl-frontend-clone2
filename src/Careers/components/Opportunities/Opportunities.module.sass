@import '@shared/stylesheets/config.sass'

$scroll-button-spacing: 40px
$scroll-button-spacing-xxl: $m-lg

.cardsWrapper
  +flex-container(column)
  flex: 1
  gap: $m-md

.scrollTopButton
  position: fixed
  right: $scroll-button-spacing
  bottom: $scroll-button-spacing
  z-index: $z-low

  +media-breakpoint-up(xxl)
    right: $scroll-button-spacing-xxl
    bottom: $scroll-button-spacing-xxl
