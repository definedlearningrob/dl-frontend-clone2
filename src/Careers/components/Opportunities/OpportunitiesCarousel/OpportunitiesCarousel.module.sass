@import '@shared/stylesheets/config.sass'

$fullHD-opportunity-card-width: 307px
$laptop-opportunity-card-width: 265px
$icon-button-space: 20px

.opportunitiesCarousel
  background-color: $color-white
  border-radius: $border-radius-sm
  padding: $m-base

  :global(.left-arrow)
    &:after
      background: $carousel-prev-arrow-gradient!important

  :global(.right-arrow)
    &:after
      background: $carousel-next-arrow-gradient !important

  +media-breakpoint-up(xxxl)
    padding: $m-md

  :global(.menu-wrapper)
    padding: 0 !important

  :global(.scroll-menu-arrow)
    +flex-container(row, center, center)
    height: 100%
    top: 0 !important

.cardHeader
  margin-bottom: $m-xxs

  +media-breakpoint-up(xxxl)
    margin-bottom: $m-xs

.opportunitiesCounter
  color: $color-neutral-600

.subHeading
  color: $color-neutral-700
  margin-bottom: $m-base

.opportunityCard
  width: $laptop-opportunity-card-width
  margin-right: $m-x !important

  +media-breakpoint-up(xxxl)
    width: $fullHD-opportunity-card-width
    margin-right: $m-base !important

.emptyState
  +flex-container(column, center, center)
