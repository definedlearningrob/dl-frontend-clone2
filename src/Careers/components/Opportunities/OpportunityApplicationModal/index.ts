export * from './OpportunityApplication';
export * from './ApplicationDetails';
export * from './ApplicationQuestions';
