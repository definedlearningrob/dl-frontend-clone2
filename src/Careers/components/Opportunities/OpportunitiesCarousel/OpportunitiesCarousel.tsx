import { useTranslation } from 'react-i18next';
import { isEmpty } from 'lodash-es';
import cx from 'classnames';

import { OpportunityCard } from '@dc/components/Opportunities/OpportunityCard';
import { ReactComponent as EmptyOpportunitiesState } from '@dc/svg/empty_opportunity_cards_section.svg';
import { useMyOpportunitiesQuery } from '@dc/graphql/student/hooks/useMyOpportunitiesQuery';
import { CarouselSkeleton } from '@dc/components/Opportunities/SkeletonOpportunities';

import SharedCarousel from '@shared/components/Carousel/Carousel';
import EmptyState from '@shared/components/EmptyState/EmptyState';

import styles from './OpportunitiesCarousel.module.sass';

export const OpportunitiesCarousel = () => {
  const { t } = useTranslation();
  const { data, loading } = useMyOpportunitiesQuery();

  if (loading) {
    return (
      <div className={styles.opportunitiesCarousel}>
        <h5 className={styles.cardHeader}>{t('opportunities.carouselHeading')}</h5>
        <p className={styles.subHeading}>{t('opportunities.carouselSubheading')}</p>
        <CarouselSkeleton count={3} />
      </div>
    );
  }

  const myOpportunities = data?.myOpportunities.nodes ?? [];

  if (isEmpty(myOpportunities)) {
    return (
      <div className={cx(styles.opportunitiesCarousel, styles.emptyState)}>
        <EmptyState
          heading={t('opportunities.emptyState.genericHeading', {
            area: t('opportunities.carouselHeading'),
          })}
          icon={<EmptyOpportunitiesState data-testid='empty-state-icon' />}>
          <p>{t('opportunities.emptyState.genericSubheading')}</p>
        </EmptyState>
      </div>
    );
  }

  const parsedData = myOpportunities.map((opportunity) => (
    <OpportunityCard key={opportunity.id} className={styles.opportunityCard} {...opportunity} />
  ));

  return (
    <div className={styles.opportunitiesCarousel}>
      <h5 className={styles.cardHeader}>
        {t('opportunities.carouselHeading')}
        <span className={styles.opportunitiesCounter}> ({myOpportunities.length})</span>
      </h5>
      <p className={styles.subHeading}>{t('opportunities.carouselSubheading')}</p>
      <SharedCarousel data={parsedData} />
    </div>
  );
};
