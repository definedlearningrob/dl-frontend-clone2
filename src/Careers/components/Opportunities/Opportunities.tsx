import { useTranslation } from 'react-i18next';
import { useCallback } from 'react';

import { OpportunitiesCarousel } from '@dc/components/Opportunities/OpportunitiesCarousel';

import { ReactComponent as ArrowUpIcon } from '@shared/svg/chevron_up.svg';
import { IconButton } from '@shared/components/IconButton/IconButton';

import { OpportunitiesListCard } from './OpportunitiesListCard';
import styles from './Opportunities.module.sass';
import { FavouriteOpportunitiesList } from './FavouriteOpportunitiesList/FavouriteOpportunitiesList';
import { PartnersList } from './PartnersList/PartnersList';

type Props = {
  hasPartners: boolean;
};

export const Opportunities = ({ hasPartners }: Props) => {
  const { t } = useTranslation();

  const handleScrollToTop = useCallback(() => {
    window.scrollTo({ top: 0, behavior: 'smooth' });
  }, []);

  return (
    <>
      <h4 className={styles.opportunitiesHeading}>{t('opportunities.heading')}</h4>
      <div className='flex flex-col gap-base xxxl:gap-md grow'>
        {hasPartners && (
          <div className='flex gap-base xxxl:gap-md h-[412px] xxxl:h-[643px]'>
            <div className='basis-[512px] xxxl:basis-[640px] shrink-0'>
              <FavouriteOpportunitiesList />
            </div>
            <div className='grow'>
              <PartnersList />
            </div>
          </div>
        )}
        {!hasPartners && <OpportunitiesCarousel />}
        <OpportunitiesListCard hasPartners={hasPartners} />
      </div>
      <IconButton
        Icon={ArrowUpIcon}
        circle={true}
        className={styles.scrollTopButton}
        size='md'
        variant='primary'
        onClick={handleScrollToTop}
      />
    </>
  );
};
