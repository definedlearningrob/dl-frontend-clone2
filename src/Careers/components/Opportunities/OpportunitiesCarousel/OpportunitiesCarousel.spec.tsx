import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { screen, waitFor } from '@testing-library/react';

import { OpportunitiesCarousel } from '@dc/components/Opportunities/OpportunitiesCarousel';
import {
  OPPORTUNITY_TYPE,
  OPPORTUNITY_APPLICATION_STATUS,
  VISIBILITY_SCOPE,
} from '@dc/resources/enums';
import { renderWithRouter } from '@dc/utils/test';
import MY_OPPORTUNITIES from '@dc/graphql/student/queries/myOpportunities';

const myOpportunitiesSpy = jest.fn();
const mockMyOpportunities = [
  {
    name: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua',
    imageUrl: 'https://picsum.photos/600/300',
    opportunityType: OPPORTUNITY_TYPE.INTERNSHIP,
    periodStart: '2023-02-03T00:00:00.000',
    periodEnd: '2023-04-03T00:00:00.000',
    deadline: '2023-03-03T00:00:00.000',
    isFavorite: true,
    isRecommended: false,
    id: '1',
    applicationStatus: OPPORTUNITY_APPLICATION_STATUS.REJECTED,
    automaticAcceptance: false,
    availableSpots: 5,
    description: 'Lorem ipsum dolor sit amet',
    location: 'Lorem Ipsum',
    tags: ['Lorem', 'ipsum', 'dolor', 'sit', 'amet'],
    visibilityScope: VISIBILITY_SCOPE.ALL,
    creditsOutcomes: '50',
    pathways: [],
    salaryInformation: '50',
    archivedAt: null,
    virtualInternship: null,
    partner: null,
    imageFitToContainer: false,
  },
  {
    name: 'Nascetur ridiculus mus mauris vitae ultricies leo integer malesuada nunc vel risus commodo viverra maecenas',
    imageUrl: 'https://picsum.photos/600/300',
    opportunityType: OPPORTUNITY_TYPE.INTERNSHIP,
    periodStart: '2023-02-03T00:00:00.000',
    periodEnd: '2023-04-03T00:00:00.000',
    deadline: '2023-03-03T00:00:00.000',
    isFavorite: true,
    isRecommended: true,
    id: '2',
    applicationStatus: OPPORTUNITY_APPLICATION_STATUS.STARTED,
    automaticAcceptance: false,
    availableSpots: 5,
    description: 'Lorem ipsum dolor sit amet',
    location: 'Lorem Ipsum',
    tags: ['Lorem', 'ipsum', 'dolor', 'sit', 'amet'],
    visibilityScope: VISIBILITY_SCOPE.ALL,
    creditsOutcomes: '50',
    pathways: [],
    salaryInformation: '50',
    archivedAt: null,
    virtualInternship: null,
    partner: null,
    imageFitToContainer: true,
  },
  {
    name: 'Libero id faucibus nisl tincidunt eget nullam non nisi est sit amet facilisis magna etiam',
    imageUrl: 'https://picsum.photos/600/300',
    opportunityType: OPPORTUNITY_TYPE.INTERNSHIP,
    periodStart: '2023-02-03T00:00:00.000',
    periodEnd: '2023-04-03T00:00:00.000',
    deadline: '2023-03-03T00:00:00.000',
    isFavorite: true,
    isRecommended: true,
    id: '3',
    applicationStatus: OPPORTUNITY_APPLICATION_STATUS.ACCEPTED,
    automaticAcceptance: false,
    availableSpots: 5,
    description: 'Lorem ipsum dolor sit amet',
    location: 'Lorem Ipsum',
    tags: ['Lorem', 'ipsum', 'dolor', 'sit', 'amet'],
    visibilityScope: VISIBILITY_SCOPE.ALL,
    creditsOutcomes: '50',
    pathways: [],
    salaryInformation: '50',
    archivedAt: null,
    virtualInternship: null,
    partner: null,
    imageFitToContainer: false,
  },
  {
    name: 'Turpis in eu mi bibendum neque egestas congue quisque egestas diam in arcu cursus euismod',
    imageUrl: 'https://picsum.photos/600/300',
    opportunityType: OPPORTUNITY_TYPE.INTERNSHIP,
    periodStart: '2023-02-03T12:00:00.000',
    periodEnd: '2023-04-03T12:00:00.000',
    deadline: '2023-03-03T12:00:00.000',
    isFavorite: true,
    isRecommended: false,
    id: '4',
    applicationStatus: OPPORTUNITY_APPLICATION_STATUS.REJECTED,
    automaticAcceptance: false,
    availableSpots: 5,
    description: 'Lorem ipsum dolor sit amet',
    location: 'Lorem Ipsum',
    tags: ['Lorem', 'ipsum', 'dolor', 'sit', 'amet'],
    visibilityScope: VISIBILITY_SCOPE.ALL,
    creditsOutcomes: '50',
    pathways: [],
    salaryInformation: '50',
    archivedAt: null,
    virtualInternship: null,
    partner: null,
    imageFitToContainer: false,
  },
];

const emptyMyOpportunityMock = [
  {
    request: {
      query: MY_OPPORTUNITIES,
      variables: { page: 1, perPage: 100 },
    },
    result: {
      data: {
        myOpportunities: {
          nodes: [],
          nodesCount: 0,
          pagesCount: 0,
        },
      },
    },
  },
];

const myOpportunityMock = [
  {
    request: {
      query: MY_OPPORTUNITIES,
      variables: { page: 1, perPage: 100 },
    },
    result: () => {
      myOpportunitiesSpy();

      return {
        data: {
          myOpportunities: {
            nodes: mockMyOpportunities,
            nodesCount: 4,
            pagesCount: 1,
          },
        },
      };
    },
  },
];

const renderOpportunityCarousel = (mocks: MockedResponse[]) =>
  renderWithRouter(
    <MockedProvider mocks={mocks}>
      <OpportunitiesCarousel />
    </MockedProvider>
  );

describe('OpportunitiesCarousel', () => {
  it('should render properly with empty state', async () => {
    const { container } = renderOpportunityCarousel(emptyMyOpportunityMock);

    expect(await screen.findByTestId('empty-state-icon')).toBeInTheDocument();
    expect(screen.getByText('Current and Favorites area is empty')).toBeInTheDocument();
    expect(screen.getByText('There is no data in this section yet.')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render properly with passed data', async () => {
    const { container } = renderOpportunityCarousel(myOpportunityMock);
    await waitFor(() => {
      expect(myOpportunitiesSpy).toHaveBeenCalled();
    });
    expect(await screen.findByText('Current and Favorites')).toBeInTheDocument();
    expect(screen.getByText(`(${mockMyOpportunities.length})`)).toBeInTheDocument();
    expect(screen.getByText('All In-progress and favorited opportunities')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
