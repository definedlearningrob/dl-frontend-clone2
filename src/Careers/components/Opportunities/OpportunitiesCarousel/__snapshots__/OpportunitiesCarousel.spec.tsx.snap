// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpportunitiesCarousel should render properly with empty state 1`] = `
<div>
  <div
    class="opportunitiesCarousel emptyState"
  >
    <div
      class="flex flex-col justify-center items-center bg-white rounded-sm height-[305px] w-full"
    >
      <div
        class="flex flex-col justify-center items-center mb-sm [&>*]:m-xxs"
      >
        <div
          data-testid="empty-state-icon"
        />
      </div>
      <h6
        class="mb-xs"
      >
        Current and Favorites area is empty
      </h6>
      <p>
        There is no data in this section yet.
      </p>
    </div>
  </div>
</div>
`;

exports[`OpportunitiesCarousel should render properly with passed data 1`] = `
<div>
  <div
    class="opportunitiesCarousel"
  >
    <h5
      class="cardHeader"
    >
      Current and Favorites
      <span
        class="opportunitiesCounter"
      >
         (
        4
        )
      </span>
    </h5>
    <p
      class="subHeading"
    >
      All In-progress and favorited opportunities
    </p>
    <div
      class="list"
    >
      <div
        class="horizontal-menu"
        style="align-items: center; display: flex; user-select: none;"
      >
        <div
          class="scroll-menu-arrow scroll-menu-arrow--disabled"
        >
          <div
            class="left-arrow relative h-full flex items-center leftArrowWrapper neutralBackgroundLeft"
          >
            <button
              class="flex items-center justify-center transition-colors rounded-sm border border-transparent cursor-pointer focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] text-primary-500 bg-white hover:bg-primary-200 w-md h-md arrow"
              data-testid="icon-button"
              type="button"
            >
              <div
                class="flex"
              >
                <div
                  class="flex align-center justify-center grow-0 shrink-0"
                  data-testid="icon"
                >
                  <div
                    class="h-base w-base"
                  />
                </div>
              </div>
            </button>
          </div>
        </div>
        <div
          class="menu-wrapper"
          style="overflow: hidden; user-select: none;"
        >
          <div
            class="menu-wrapper--inner"
            style="transform: translate3d(0px, 0, 0); transition: transform 0.4s; white-space: nowrap; text-align: left;"
          >
            <div
              class="menu-item-wrapper "
              role="button"
              style="display: inline-block;"
              tabindex="0"
            >
              <a
                aria-label="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua"
                class="bg-white rounded-sm border border-neutral-300 text-font-primary transition-[box-shadow] hover:shadow-300 hover:text-font-primary group/card focus-visible:outline focus-visible:outline-1 focus-visible:outline-primary-500 card notAccepted opportunityCard verticalCard"
                href="/opportunities/1"
              >
                <div
                  class="relative"
                >
                  <div
                    class="rounded-t-sm overflow-hidden shrink-0 grow-0 before:absolute before:bg-neutral-800/[32%] before:inset-0 before:group-focus-visible/card:!bg-neutral-800/[16%] before:group-hover/card:!bg-neutral-800/[16%] aspect-[2.45/1] xxxl:aspect-[2.54/1] relative"
                  >
                    <button
                      class="flex items-center justify-center transition-colors rounded-sm border border-transparent cursor-pointer focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] text-primary-500 bg-white hover:bg-primary-200 h-base w-base  iconButton favoriteButton focus-visible:outline focus-visible:!outline-danger-600 focus-visible:!outline-1 focus-visible:!outline-offset-2 md"
                      data-testid="icon-button"
                      type="button"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                    </button>
                    <img
                      alt="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua"
                      class="h-full mx-auto object-cover"
                      src="https://picsum.photos/600/300"
                    />
                  </div>
                </div>
                <div
                  class="cardContent"
                >
                  <section
                    class="body p-x xxxl:p-sm"
                  >
                    <div
                      class="flex justify-between gap-xxs mb-x xxxl:mb-sm"
                    >
                      <div>
                        <span
                          data-state="closed"
                        >
                          <h6
                            class="line-clamp-2 text-xs mb-0 whitespace-break-spaces"
                          >
                            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua
                          </h6>
                        </span>
                      </div>
                      <div
                        class="font-medium leading-lg flex items-center whitespace-nowrap self-start bg-danger-100 text-danger-500 p-xxxs text-xxs rounded-xs gap-xxxs"
                      >
                        Not accepted
                      </div>
                    </div>
                    <span
                      class="metadata"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0 text-font-secondary mr-xxs xxxl:mr-xs"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                      <span
                        class="metadataLabel"
                      >
                        Type:
                      </span>
                      Internship
                    </span>
                    <span
                      class="flex items-center text-xxs mb-xxs last:mb-0 xxxl:text-xs"
                    >
                      <div
                        class="icon text-font-secondary mr-xxs xxxl:mr-xs -xs-size"
                        data-testid="icon"
                      >
                        <div />
                      </div>
                      <span
                        class="text-font-secondary mr-xxs"
                      >
                        Period:
                      </span>
                      <span
                        data-state="closed"
                      >
                        <span
                          class="line-clamp-1"
                        >
                          Feb 3, 2023 - Apr 3, 2023
                        </span>
                      </span>
                    </span>
                    <span
                      class="flex items-center text-xxs mb-xxs last:mb-0 xxxl:text-xs"
                    >
                      <div
                        class="icon text-font-secondary mr-xxs xxxl:mr-xs -xs-size"
                        data-testid="icon"
                      >
                        <div />
                      </div>
                      <span
                        class="text-font-secondary mr-xxs"
                      >
                        Deadline:
                      </span>
                      Mar 3, 2023
                    </span>
                  </section>
                  <div
                    class="footer px-sm py-x"
                  >
                    Learn more
                    <div
                      class="flex"
                    >
                      <div
                        class="flex align-center justify-center grow-0 shrink-0"
                        data-testid="icon"
                      >
                        <div
                          class="h-sm w-sm"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </a>
            </div>
            <div
              class="menu-item-wrapper "
              role="button"
              style="display: inline-block;"
              tabindex="0"
            >
              <a
                aria-label="Nascetur ridiculus mus mauris vitae ultricies leo integer malesuada nunc vel risus commodo viverra maecenas"
                class="bg-white rounded-sm border border-neutral-300 text-font-primary transition-[box-shadow] hover:shadow-300 hover:text-font-primary group/card focus-visible:outline focus-visible:outline-1 focus-visible:outline-primary-500 card inProgress opportunityCard verticalCard"
                href="/opportunities/2"
              >
                <div
                  class="relative"
                >
                  <div
                    class="rounded-t-sm overflow-hidden shrink-0 grow-0 before:absolute before:bg-neutral-800/[32%] before:inset-0 before:group-focus-visible/card:!bg-neutral-800/[16%] before:group-hover/card:!bg-neutral-800/[16%] aspect-[2.45/1] xxxl:aspect-[2.54/1] relative"
                  >
                    <button
                      class="flex items-center justify-center transition-colors rounded-sm border border-transparent cursor-pointer focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] text-primary-500 bg-white hover:bg-primary-200 h-base w-base  iconButton favoriteButton focus-visible:outline focus-visible:!outline-danger-600 focus-visible:!outline-1 focus-visible:!outline-offset-2 md"
                      data-testid="icon-button"
                      type="button"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                    </button>
                    <img
                      alt="Nascetur ridiculus mus mauris vitae ultricies leo integer malesuada nunc vel risus commodo viverra maecenas"
                      class="object-cover w-full h-full"
                      src="https://picsum.photos/600/300"
                    />
                  </div>
                </div>
                <div
                  class="cardContent"
                >
                  <section
                    class="body p-x xxxl:p-sm"
                  >
                    <div
                      class="flex justify-between gap-xxs mb-x xxxl:mb-sm"
                    >
                      <div>
                        <span
                          data-state="closed"
                        >
                          <h6
                            class="line-clamp-2 text-xs mb-0 whitespace-break-spaces"
                          >
                            Nascetur ridiculus mus mauris vitae ultricies leo integer malesuada nunc vel risus commodo viverra maecenas
                          </h6>
                        </span>
                      </div>
                      <div
                        class="font-medium leading-lg flex items-center whitespace-nowrap self-start bg-secondary-200 text-secondary-500 p-xxxs text-xxs rounded-xs gap-xxxs"
                      >
                        In-progress
                      </div>
                    </div>
                    <span
                      class="metadata"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0 text-font-secondary mr-xxs xxxl:mr-xs"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                      <span
                        class="metadataLabel"
                      >
                        Type:
                      </span>
                      Internship
                    </span>
                    <span
                      class="flex items-center text-xxs mb-xxs last:mb-0 xxxl:text-xs"
                    >
                      <div
                        class="icon text-font-secondary mr-xxs xxxl:mr-xs -xs-size"
                        data-testid="icon"
                      >
                        <div />
                      </div>
                      <span
                        class="text-font-secondary mr-xxs"
                      >
                        Period:
                      </span>
                      <span
                        data-state="closed"
                      >
                        <span
                          class="line-clamp-1"
                        >
                          Feb 3, 2023 - Apr 3, 2023
                        </span>
                      </span>
                    </span>
                    <span
                      class="flex items-center text-xxs mb-xxs last:mb-0 xxxl:text-xs"
                    >
                      <div
                        class="icon text-font-secondary mr-xxs xxxl:mr-xs -xs-size"
                        data-testid="icon"
                      >
                        <div />
                      </div>
                      <span
                        class="text-font-secondary mr-xxs"
                      >
                        Deadline:
                      </span>
                      Mar 3, 2023
                    </span>
                  </section>
                  <div
                    class="footer px-sm py-x"
                  >
                    Continue
                    <div
                      class="flex"
                    >
                      <div
                        class="flex align-center justify-center grow-0 shrink-0"
                        data-testid="icon"
                      >
                        <div
                          class="h-sm w-sm"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </a>
            </div>
            <div
              class="menu-item-wrapper "
              role="button"
              style="display: inline-block;"
              tabindex="0"
            >
              <a
                aria-label="Libero id faucibus nisl tincidunt eget nullam non nisi est sit amet facilisis magna etiam"
                class="bg-white rounded-sm border border-neutral-300 text-font-primary transition-[box-shadow] hover:shadow-300 hover:text-font-primary group/card focus-visible:outline focus-visible:outline-1 focus-visible:outline-primary-500 card accepted opportunityCard verticalCard"
                href="/opportunities/3"
              >
                <div
                  class="relative"
                >
                  <div
                    class="rounded-t-sm overflow-hidden shrink-0 grow-0 before:absolute before:bg-neutral-800/[32%] before:inset-0 before:group-focus-visible/card:!bg-neutral-800/[16%] before:group-hover/card:!bg-neutral-800/[16%] aspect-[2.45/1] xxxl:aspect-[2.54/1] relative"
                  >
                    <button
                      class="flex items-center justify-center transition-colors rounded-sm border border-transparent cursor-pointer focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] text-primary-500 bg-white hover:bg-primary-200 h-base w-base  iconButton favoriteButton focus-visible:outline focus-visible:!outline-danger-600 focus-visible:!outline-1 focus-visible:!outline-offset-2 md"
                      data-testid="icon-button"
                      type="button"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                    </button>
                    <img
                      alt="Libero id faucibus nisl tincidunt eget nullam non nisi est sit amet facilisis magna etiam"
                      class="h-full mx-auto object-cover"
                      src="https://picsum.photos/600/300"
                    />
                  </div>
                </div>
                <div
                  class="cardContent"
                >
                  <section
                    class="body p-x xxxl:p-sm"
                  >
                    <div
                      class="flex justify-between gap-xxs mb-x xxxl:mb-sm"
                    >
                      <div>
                        <span
                          data-state="closed"
                        >
                          <h6
                            class="line-clamp-2 text-xs mb-0 whitespace-break-spaces"
                          >
                            Libero id faucibus nisl tincidunt eget nullam non nisi est sit amet facilisis magna etiam
                          </h6>
                        </span>
                      </div>
                      <div
                        class="font-medium leading-lg flex items-center whitespace-nowrap self-start bg-success-100 text-success-500 p-xxxs text-xxs rounded-xs gap-xxxs"
                      >
                        Accepted
                      </div>
                    </div>
                    <span
                      class="metadata"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0 text-font-secondary mr-xxs xxxl:mr-xs"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                      <span
                        class="metadataLabel"
                      >
                        Type:
                      </span>
                      Internship
                    </span>
                    <span
                      class="flex items-center text-xxs mb-xxs last:mb-0 xxxl:text-xs"
                    >
                      <div
                        class="icon text-font-secondary mr-xxs xxxl:mr-xs -xs-size"
                        data-testid="icon"
                      >
                        <div />
                      </div>
                      <span
                        class="text-font-secondary mr-xxs"
                      >
                        Period:
                      </span>
                      <span
                        data-state="closed"
                      >
                        <span
                          class="line-clamp-1"
                        >
                          Feb 3, 2023 - Apr 3, 2023
                        </span>
                      </span>
                    </span>
                    <span
                      class="flex items-center text-xxs mb-xxs last:mb-0 xxxl:text-xs"
                    >
                      <div
                        class="icon text-font-secondary mr-xxs xxxl:mr-xs -xs-size"
                        data-testid="icon"
                      >
                        <div />
                      </div>
                      <span
                        class="text-font-secondary mr-xxs"
                      >
                        Deadline:
                      </span>
                      Mar 3, 2023
                    </span>
                  </section>
                  <div
                    class="footer px-sm py-x"
                  >
                    Learn more
                    <div
                      class="flex"
                    >
                      <div
                        class="flex align-center justify-center grow-0 shrink-0"
                        data-testid="icon"
                      >
                        <div
                          class="h-sm w-sm"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </a>
            </div>
            <div
              class="menu-item-wrapper "
              role="button"
              style="display: inline-block;"
              tabindex="0"
            >
              <a
                aria-label="Turpis in eu mi bibendum neque egestas congue quisque egestas diam in arcu cursus euismod"
                class="bg-white rounded-sm border border-neutral-300 text-font-primary transition-[box-shadow] hover:shadow-300 hover:text-font-primary group/card focus-visible:outline focus-visible:outline-1 focus-visible:outline-primary-500 card notAccepted opportunityCard verticalCard"
                href="/opportunities/4"
              >
                <div
                  class="relative"
                >
                  <div
                    class="rounded-t-sm overflow-hidden shrink-0 grow-0 before:absolute before:bg-neutral-800/[32%] before:inset-0 before:group-focus-visible/card:!bg-neutral-800/[16%] before:group-hover/card:!bg-neutral-800/[16%] aspect-[2.45/1] xxxl:aspect-[2.54/1] relative"
                  >
                    <button
                      class="flex items-center justify-center transition-colors rounded-sm border border-transparent cursor-pointer focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] text-primary-500 bg-white hover:bg-primary-200 h-base w-base  iconButton favoriteButton focus-visible:outline focus-visible:!outline-danger-600 focus-visible:!outline-1 focus-visible:!outline-offset-2 md"
                      data-testid="icon-button"
                      type="button"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                    </button>
                    <img
                      alt="Turpis in eu mi bibendum neque egestas congue quisque egestas diam in arcu cursus euismod"
                      class="h-full mx-auto object-cover"
                      src="https://picsum.photos/600/300"
                    />
                  </div>
                </div>
                <div
                  class="cardContent"
                >
                  <section
                    class="body p-x xxxl:p-sm"
                  >
                    <div
                      class="flex justify-between gap-xxs mb-x xxxl:mb-sm"
                    >
                      <div>
                        <span
                          data-state="closed"
                        >
                          <h6
                            class="line-clamp-2 text-xs mb-0 whitespace-break-spaces"
                          >
                            Turpis in eu mi bibendum neque egestas congue quisque egestas diam in arcu cursus euismod
                          </h6>
                        </span>
                      </div>
                      <div
                        class="font-medium leading-lg flex items-center whitespace-nowrap self-start bg-danger-100 text-danger-500 p-xxxs text-xxs rounded-xs gap-xxxs"
                      >
                        Not accepted
                      </div>
                    </div>
                    <span
                      class="metadata"
                    >
                      <div
                        class="flex"
                      >
                        <div
                          class="flex align-center justify-center grow-0 shrink-0 text-font-secondary mr-xxs xxxl:mr-xs"
                          data-testid="icon"
                        >
                          <div
                            class="h-sm w-sm"
                          />
                        </div>
                      </div>
                      <span
                        class="metadataLabel"
                      >
                        Type:
                      </span>
                      Internship
                    </span>
                    <span
                      class="flex items-center text-xxs mb-xxs last:mb-0 xxxl:text-xs"
                    >
                      <div
                        class="icon text-font-secondary mr-xxs xxxl:mr-xs -xs-size"
                        data-testid="icon"
                      >
                        <div />
                      </div>
                      <span
                        class="text-font-secondary mr-xxs"
                      >
                        Period:
                      </span>
                      <span
                        data-state="closed"
                      >
                        <span
                          class="line-clamp-1"
                        >
                          Feb 3, 2023 - Apr 3, 2023
                        </span>
                      </span>
                    </span>
                    <span
                      class="flex items-center text-xxs mb-xxs last:mb-0 xxxl:text-xs"
                    >
                      <div
                        class="icon text-font-secondary mr-xxs xxxl:mr-xs -xs-size"
                        data-testid="icon"
                      >
                        <div />
                      </div>
                      <span
                        class="text-font-secondary mr-xxs"
                      >
                        Deadline:
                      </span>
                      Mar 3, 2023
                    </span>
                  </section>
                  <div
                    class="footer px-sm py-x"
                  >
                    Learn more
                    <div
                      class="flex"
                    >
                      <div
                        class="flex align-center justify-center grow-0 shrink-0"
                        data-testid="icon"
                      >
                        <div
                          class="h-sm w-sm"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </a>
            </div>
          </div>
        </div>
        <div
          class="scroll-menu-arrow scroll-menu-arrow--disabled"
        >
          <div
            class="right-arrow relative h-full flex items-center rightArrowWrapper neutralBackgroundRight"
          >
            <button
              class="flex items-center justify-center transition-colors rounded-sm border border-transparent cursor-pointer focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] text-primary-500 bg-white hover:bg-primary-200 w-md h-md arrow"
              data-testid="icon-button"
              type="button"
            >
              <div
                class="flex"
              >
                <div
                  class="flex align-center justify-center grow-0 shrink-0"
                  data-testid="icon"
                >
                  <div
                    class="h-base w-base"
                  />
                </div>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
